test_data/asm/generic_permutation.asm

```diff
@@ -0,0 +1,37 @@
+let N: int = 8;
+
+use std::array;
+
+
+let permutation: expr[], expr[] -> = constr |left, right| {
+    std::check::assert_eq(left.len(), right.len());
+
+    let alpha: expr = std::prover::challenge(0, 1);
+    let beta: expr = std::prover::challenge(0, 2);
+
+    let left_compressed = compress_selected_exprs(id, alpha, beta, 1, left);
+    let right_compressed = compress_selected_exprs(id, alpha, beta, 1, right);
+
+    let z = std::prover::new_witness_in_stage(1);
+
+    let first: col = [1] + [0]*;
+
+    first * (z - 1) = 0;
+
+    // Update rule:
+    // z' = z * (beta - a) / (beta - b)
+    (beta - left_compressed) * z' = z * (beta - right_compressed);
+};
+
+let compress_selected_exprs: int, expr, expr, expr, expr[] -> expr = |id, alpha, beta, sel, exprs|
+    sel * (compress_columns(alpha, exprs) + id + beta - 1) + 1;
```

Comment:
Where does the ID come from? We don't really need it, do we? I think it's a way to share the accumulator between several permutations, which we're not doing here?